tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/MsTestIntegration.cs

```diff
@@ -23,75 +24,83 @@ internal static class MsTestIntegration
+        var testMethodArguments = testMethodInstance.Arguments;
+        var testName = testMethodInstance.TestMethodName;
 
-        if (TestSuite.Current is { } suite)
+        var suite = TestSuite.Current;
+        if (suite is null && testMethodInstance.Instance.TryDuckCast<ITestMethodInfo>(out var testMethodInfo))
         {
-            var test = suite.InternalCreateTest(testName);
+            suite = GetOrCreateTestSuiteFromTestClassInfo(testMethodInfo.Parent);
+        }
 
-            // Get test parameters
-            var methodParameters = testMethod.GetParameters();
-            if (methodParameters?.Length > 0)
-            {
-                var testParameters = new TestParameters();
-                testParameters.Metadata = new Dictionary<string, object>();
-                testParameters.Arguments = new Dictionary<string, object>();
+        if (suite is null)
+        {
+            Log.Warning("There's no suite to create the Test instance.");
+            return null;
+        }
 
-                for (int i = 0; i < methodParameters.Length; i++)
-                {
-                    if (testMethodArguments != null && i < testMethodArguments.Length)
-                    {
-                        testParameters.Arguments[methodParameters[i].Name] = Common.GetParametersValueData(testMethodArguments[i]);
-                    }
-                    else
-                    {
-                        testParameters.Arguments[methodParameters[i].Name] = "(default)";
-                    }
-                }
+        var test = suite.InternalCreateTest(testName);
 
-                test.SetParameters(testParameters);
-            }
+        // Get test parameters
+        var methodParameters = testMethod.GetParameters();
+        if (methodParameters?.Length > 0)
+        {
+            var testParameters = new TestParameters
+            {
+                Metadata = new Dictionary<string, object>(),
+                Arguments = new Dictionary<string, object>()
+            };
 
-            // Get traits
-            if (GetTraits(testMethod) is { } testTraits)
+            for (var i = 0; i < methodParameters.Length; i++)
             {
-                // Unskippable tests
-                if (CIVisibility.Settings.IntelligentTestRunnerEnabled)
+                if (testMethodArguments != null && i < testMethodArguments.Length)
                 {
-                    ShouldSkip(testMethodInfo, out var isUnskippable, out var isForcedRun, testTraits);
-                    test.SetTag(IntelligentTestRunnerTags.UnskippableTag, isUnskippable ? "true" : "false");
-                    test.SetTag(IntelligentTestRunnerTags.ForcedRunTag, isForcedRun ? "true" : "false");
-                    testTraits.Remove(IntelligentTestRunnerTags.UnskippableTraitName);
+                    testParameters.Arguments[methodParameters[i].Name ?? string.Empty] = Common.GetParametersValueData(testMethodArguments[i]);
```

Comment:
If more than one of `methodParameters[i].Name` returns null, then we overwrite the same argument multiple times.

Does it make more sense to use `$"{i}"` as the argument name, that way the names are unique, but aren't valid identifiers, so shouldn't get conflicts? 🤔